worker.go

```diff
@@ -247,7 +241,11 @@ func DetectContentFromReader(reader io.ReadCloser, format, name string, parent *
 		log.WithFields(log.Fields{logLayerName: name, "component count": len(m.components)}).Debug("detected components")
 	}
 
-	namespace, distroless, features, rhelv2Components, languageComponents, err := detectFromFiles(*files, name, parent, m.components, uncertifiedRHEL)
+	namespace, features, rhelv2Components, languageComponents, err := detectFromFiles(*files, name, parent, m.components, uncertifiedRHEL)
+	distroless := isDistroless(*files) || (parent != nil && parent.Distroless)
+	if !env.LanguageVulns.Enabled() {
```

Comment:
Maybe not in this PR, but it seems like we're going to do a bunch of work to find language components and store a bunch of data just to toss them later. I think this check should probably be much earlier

Reply:
Agreed, I can actually do it in a different PR.